nixops/resources/datadog-monitor.py

```diff
@@ -0,0 +1,193 @@
+
+    @classmethod
+    def get_resource_type(cls):
+        return "datadogMonitors"
+
+    def __init__(self, xml):
+        nixops.resources.ResourceDefinition.__init__(self, xml)
+        self.extraOptions = {}
+        self.extraOptions['thresholds']= {};
+        self.monitorName = xml.find("attrs/attr[@name='name']/string").get("value")
+        self.monitorType = xml.find("attrs/attr[@name='type']/string").get("value")
+        self.query = xml.find("attrs/attr[@name='query']/string").get("value")
+        self.apiKey = xml.find("attrs/attr[@name='apiKey']/string").get("value")
+        self.appKey = xml.find("attrs/attr[@name='appKey']/string").get("value")
+        self.message = xml.find("attrs/attr[@name='message']/string").get("value")
+        self.setOptionalAttr(xml=xml, name='renotifyInterval', api_name="renotify_interval", type='int')
+        self.setOptionalAttr(xml=xml, name='silenced', api_name='silenced', type='string')
+        self.setOptionalAttr(xml=xml, name='escalationMessage', api_name='escalation_message', type='string')
+        self.setOptionalAttr(xml=xml, name='notifyNoData', api_name='notify_no_data', type='bool')
+        self.setOptionalAttr(xml=xml, name='noDataTimeframe', api_name='no_data_timeframe', type='int')
+        self.setOptionalAttr(xml=xml, name='timeoutH', api_name='timeout_h', type='int')
+        self.setOptionalAttr(xml=xml, name='requireFullWindow', api_name='require_full_window', type='bool')
+        self.setOptionalAttr(xml=xml, name='notifyAudit', api_name='notify_audit', type='bool')
+        self.setOptionalAttr(xml=xml, name='locked', api_name='locked', type='bool')
+        self.setOptionalAttr(xml=xml, name='includeTags', api_name='include_tags', type='bool')
+        for alert in xml.findall("attrs/attr[@name='thresholds']/attrs/attr"):
+            if alert.find("int") != None:
+                if alert.attrib.get('name') == "ok":
+                    self.extraOptions['thresholds']['ok'] = int(alert.find("int").get("value"))
+                if alert.attrib.get('name') == "critical":
+                    self.extraOptions['thresholds']['critical'] = int(alert.find("int").get("value"))
+                if alert.attrib.get('name') == "warning":
+                    self.extraOptions['thresholds']['warning'] = int(alert.find("int").get("value"))
+
+    def setOptionalAttr(self,xml,name,api_name,type):
+        attr = xml.find("attrs/attr[@name='{0}']/{1}".format(name,type))
+        if attr != None:
+            value = attr.get('value')
+            self.extraOptions[api_name]= ast.literal_eval(value) if name == 'silenced' else value
+
+    def show_type(self):
+        return "{0}".format(self.get_type())
+
+class DatadogMonitorState(nixops.resources.ResourceState):
+    """State of a Datadog monitor"""
+
+    state = nixops.util.attr_property("state", nixops.resources.ResourceState.MISSING, int)
+    apiKey = nixops.util.attr_property("apiKey", None)
```

Comment:
In python code, we usually use _'s and not camelCase for properties of python classes.